UnitTests/BraintreeSEPADirectDebitTests/SEPADirectDebitAPI_Tests.swift

```diff
@@ -0,0 +1,91 @@
+import XCTest
+@testable import BraintreeCore
+@testable import BraintreeSEPADirectDebit
+
+class SEPADirectDebitAPI_Tests: XCTestCase {
+    var billingAddress = BTPostalAddress()
+    var sepaDirectDebitRequest = BTSEPADirectDebitRequest()
+
+    override func setUp() {
+        billingAddress.streetAddress = "Kantstraße 70"
+        billingAddress.extendedAddress = "#170"
+        billingAddress.locality = "Freistaat Sachsen"
+        billingAddress.region = "Annaberg-buchholz"
+        billingAddress.postalCode = "09456"
+        billingAddress.countryCodeAlpha2 = "FR"
+        
+        sepaDirectDebitRequest.accountHolderName = "John Doe"
+        sepaDirectDebitRequest.iban = "FR891751244434203564412313"
+        sepaDirectDebitRequest.customerID = "A0E243A0A200491D929D"
+        sepaDirectDebitRequest.mandateType = .oneOff
+        sepaDirectDebitRequest.billingAddress = billingAddress
+        sepaDirectDebitRequest.merchantAccountID = "eur_pwpp_multi_account_merchant_account"
+    }
+
+    func testBuildHttpRequest_withAllParams() {
+        let expectedHTTPBody = """
+{\"sepa_debit\":{\"customer_id\":\"A0E243A0A200491D929D\",\"mandate_type\":\"ONE_OFF\",\"account_holder_name\":\"John Doe\",\"iban\":\"FR891751244434203564412313\",\"billing_address\":{\"admin_area_2\":\"Annaberg-buchholz\",\"country_code\":\"FR\",\"address_line_2\":\"#170\",\"address_line_1\":\"Kantstraße 70\",\"admin_area_1\":\"Freistaat Sachsen\",\"postal_code\":\"09456\"}},\"merchant_account_id\":\"eur_pwpp_multi_account_merchant_account\",\"cancel_url\":\"https:\\/\\/example.com\",\"return_url\":\"https:\\/\\/example.com\"}
```

Comment:
nit: indentation

Reply:
The test fails with any syntax other than this because of how the object is compiled. Let me dig into it and see if there is a different way to go about this because I agree, this is _ugly_